client/product_details/ExpandedView.jsx

```diff
@@ -1,24 +1,94 @@
 import React from 'react';
 
-const ExpandedView = (props) => {
+class ExpandedView extends React.Component {
+  constructor(props) {
+    super(props);
 
-  if (props.selectedStyle) {
-    var photoGallery = props.selectedStyle.photos;
-    return (
-      <div id="expandedView">
-        <a target="_blank">
-          <img 
-            src={photoGallery[props.currPhotoIndex].url || 'https://www.thermaxglobal.com/wp-content/uploads/2020/05/image-not-found.jpg'} 
-            alt={props.selectedStyle.name} 
-            className="defaultView" 
-            id={props.selectedStyle.style_id}>
-          </img>
-          <a onClick={props.toggleView} id="fullscreenIcon"><i class="fas fa-expand"></i></a>
-        </a>
-      </div>
-    );
-  } 
-  return null;
-};
+    this.state = {
+      zoomView: false
+    };
+    
+    this.toggleZoomView = this.toggleZoomView.bind(this);
+    this.handleMouseMove = this.handleMouseMove.bind(this);
+  }
+  
+  componentDidMount() {
+    var element = document.getElementById('expandedView');
+    console.log(element);
+    element.addEventListener("mousemove", this.handleMouseMove);
+  }
+
+  componentWillUnmount() {
+    var element = document.getElementById('expandedView');
+    console.log(element);
+    element.removeEventListener("mousemove", this.handleMouseMove);
+  }
+
+  toggleZoomView() {
+    this.setState({
+      zoomView: !this.state.zoomView
+    });
+  }
+  
+  handleMouseMove(e) {
+    console.log('in mouse move');
+    var element = document.getElementById('expandedView');
+    console.log('height: ', element.clientHeight);
+    console.log('width: ', element.clientWidth);
+    console.log('e background X: ', element.style.backgroundPositionX);
+    console.log('e background Y: ', element.style.backgroundPositionY);
+    console.log('e offset X: ', -e.offsetX);
+    console.log('e offset Y: ', -e.offsetY);
+    console.log('x%: ', (e.offsetX / element.clientWidth) * 100);
+    console.log('y%: ', (e.offsetY / element.clientHeight) * 100);
+    console.log('background size: ', element.style.backgroundSize);
+    var xPercent = (e.offsetX / element.clientWidth) * 100;
+    var yPercent = (e.offsetY / element.clientHeight) * 100;
+    element.style.backgroundPositionX = xPercent + "%";
+    element.style.backgroundPositionY = yPercent + "%";
+  }
+
+  render() {
+    if (this.props.selectedStyle && !this.state.zoomView) {
+      var photoGallery = this.props.selectedStyle.photos;
+      return (
+        <div id="expandedView">
+          <a target="_blank">
+            <img
+              onClick={this.toggleZoomView}
+              src={photoGallery[this.props.currPhotoIndex].url} 
+              alt={this.props.selectedStyle.name} 
```

Comment:
Just want to add that it's great that you take care of the "alt" for all the images! A great reminder for myself that I need to do it as well.